spinalcordtoolbox/__init__.py

```diff
@@ -1,3 +1,31 @@
 #!/usr/bin/env python
+import os
+import multiprocessing
 
 from .utils import __version__, __sct_dir__, __data_dir__, __deepseg_dir__
+
+
+def configure_sct_env_variables():
+    """
+    Set environment variables that should be active in the scope of the running Python process.
+
+    Note: This change will be active whenever `spinalcordtoolbox` is imported. So, try to only make light
+    changes here, as to not significantly impact any downstream packages built on `spinalcordtoolbox`.
+    """
+    # This is used by ANTs tools (e.g. isct_antsSliceRegularization) to control multiprocessing
+    if "ITK_GLOBAL_DEFAULT_NUMBER_OF_THREADS" not in os.environ:
+        os.environ["ITK_GLOBAL_DEFAULT_NUMBER_OF_THREADS"] = str(multiprocessing.cpu_count())
+
+    # DISPLAY is used by the X Window System (in UNIX): https://docstore.mik.ua/orelly/unix3/upt/ch35_08.htm
+    # We use this environment variable to detect whether or not we're on a headless system.
+    # NB: Sometimes 'DISPLAY' can be unset for SSH sessions that aren't headless. If any users encounter issues
+    # displaying plots, direct them to https://unix.stackexchange.com/q/138936
+    if "DISPLAY" not in os.environ:
+        # If we're on a headless system, set matplotlib's backend to 'Agg', which is a non-interactive backend.
+        # This will prevent interactive plots from being shown, which keeps headless systems from hanging indefinitely.
+        os.environ["MPLBACKEND"] = "Agg"
```

Comment:
> Is that expected behaviour for this PR?

@jcohenadad No, sadly. I believe this is due to a detail @valosekj noticed in https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/3392#issuecomment-846515398. 

> Problem can be that [MacOS doesn't run X11 by default](https://stackoverflow.com/a/31622440/12605960)

Currently we use an X11-specific environment variable (`DISPLAY`) to determine whether the system is headless. But, given macOS doesn't use X11, `DISPLAY` isn't set, causing a false positive.

Reply:
We could do something like:

```python
if sys.platform == "posix" and "DISPLAY" not in os.environment:
    os.environ["MPLBACKEND"] = "Agg"
```

But then we would need to find an alternate way of detecting headless macOS machines.

Reply:
Looking into this a little more, there's an argument that trying to detect headless machines [isn't the right soltuion](https://stackoverflow.com/a/52966593). 

Still, there are environment-specific steps for detecting this in [Java](https://stackoverflow.com/a/52987766) which may be worth keeping in mind.

Reply:
Done in eab23cb and 02b69e7.

I've also removed the test I wrote. Because `DISPLAY` is not set on macOS, this test would run and fail. At that point, it's not a very useful test.

(The test was still useful for debugging, though.)